native_client/swift/stt_ios_test/SpeechRecognitionImpl.swift

```diff
@@ -8,149 +8,48 @@
 //  Copyright © 2021 Coqui GmbH
 import Foundation
 import AVFoundation
-import AudioToolbox
 import Accelerate
 
 import stt_ios
 
-struct FillComplexInputParm {
-    var source: UnsafeMutablePointer<Int8>
-    var sourceSize: UInt32
-};
-
-class SpeechRecognitionImpl : NSObject, AVCaptureAudioDataOutputSampleBufferDelegate {
+class SpeechRecognitionImpl {
     private var model: STTModel
     private var stream: STTStream?
 
-    private var captureSession = AVCaptureSession()
     private var audioData = Data()
+    private var audioInput: AudioInput? = nil
 
-    override init() {
+    init() {
         let modelPath = Bundle.main.path(forResource: "coqui-stt-0.9.3-models", ofType: "tflite")!
         let scorerPath = Bundle.main.path(forResource: "coqui-stt-0.9.3-models", ofType: "scorer")!
 
         model = try! STTModel(modelPath: modelPath)
         try! model.enableExternalScorer(scorerPath: scorerPath)
-
-        super.init()
-
-        // prepare audio capture
-        self.configureCaptureSession()
-    }
-
-    // MARK: Microphone recognition
-
-    private func configureCaptureSession() {
-        captureSession.beginConfiguration()
-
-        let audioDevice = AVCaptureDevice.default(.builtInMicrophone, for: .audio, position: .unspecified)
-
-        let audioDeviceInput = try! AVCaptureDeviceInput(device: audioDevice!)
-        guard captureSession.canAddInput(audioDeviceInput) else { return }
-        captureSession.addInput(audioDeviceInput)
-
-        let serialQueue = DispatchQueue(label: "serialQueue")
-        let audioOutput = AVCaptureAudioDataOutput()
-        audioOutput.setSampleBufferDelegate(self, queue: serialQueue)
-
-        guard captureSession.canAddOutput(audioOutput) else { return }
-        captureSession.sessionPreset = .inputPriority
-        captureSession.addOutput(audioOutput)
-        captureSession.commitConfiguration()
     }
 
-    func captureOutput(_ output: AVCaptureOutput, didOutput sampleBuffer: CMSampleBuffer, from connection: AVCaptureConnection) {
-        var sourceFormat = (sampleBuffer.formatDescription?.audioFormatList[0].mASBD)!
-        var destinationFormat = sourceFormat
-        destinationFormat.mSampleRate = 16000.0
-
-        var audioConverterRef: AudioConverterRef?
-        let createConverterStatus = AudioConverterNew(&sourceFormat, &destinationFormat, &audioConverterRef)
+    public func startMicrophoneRecognition() {
+        audioData = Data()
+        stream = try! model.createStream()
 
-        if (createConverterStatus != noErr) {
-            print("Error creating converter")
+        audioInput = AudioInput() { shorts in
+            self.stream!.feedAudioContent(buffer: shorts)
+            self.audioData.append(shorts)
+            let partialResult = self.stream!.intermediateDecode()
+            print(partialResult)
```

Comment:
With the 2048 samples of buffering this is decoding every 128ms which is quite excessive. The old code also didn't have this printing at all. I'd prefer if this was either slower (at least every 360ms) which is the size of a full forward pass of 16 timesteps, or removed.

Reply:
I see. Regarding the buffer size, it's a little bit tricky to know exactly the interval in which the decoding will run, because `installTap` will use the sample rate of the input format, which is dependant on the microphone that's being used (e.g. my Mac 48000Hz). But those possibly large sample rates make it even worse. I could either just triple the buffer size or calculate the buffer size from a target interval (e.g. 360ms) and the input format sample rate.

Reply:
E.g. `bufferSize: inputFormat.sampleRate * processingInterval`

Reply:
It would be better to decouple the intermediate decoding from the buffer size (otherwise you're adding latency into the pipeline just to control for this).

Reply:
How would you do this? Is there some code I could look into?

Reply:
You can either use some timer facility in Swift or simply count the number of fed samples for a stream and trigger the intermediate decode once `fedSamples >= inputFormat.sampleRate * processingInterval`. Then reset `fedSamples = 0` and keep going.